client/landing/stepper/declarative-flow/videopress-tv-purchase.ts

```diff
@@ -0,0 +1,202 @@
+import { useSelect, useDispatch } from '@wordpress/data';
+import { translate } from 'i18n-calypso';
+import { useEffect, useState } from 'react';
+import { useSupportedPlans } from 'calypso/../packages/plans-grid/src/hooks';
+import { cartManagerClient } from 'calypso/my-sites/checkout/cart-manager-client';
+import {
+	setSignupCompleteSlug,
+	persistSignupDestination,
+	setSignupCompleteFlowName,
+} from 'calypso/signup/storageUtils';
+import { useSiteSlug } from '../hooks/use-site-slug';
+import { PLANS_STORE, SITE_STORE, USER_STORE, ONBOARD_STORE } from '../stores';
+import './internals/videopress.scss';
+import ProcessingStep from './internals/steps-repository/processing-step';
+import VideoPressTvRedirectStep from './internals/steps-repository/videopress-tv-redirect';
+import type { Flow, ProvidedDependencies } from './internals/types';
+import type { UserSelect, SiteSelect, PlansSelect } from '@automattic/data-stores';
+import type { MinimalRequestCartProduct } from '@automattic/shopping-cart';
+
+const videopressTvPurchase: Flow = {
+	name: VIDEOPRESS_TV_PURCHASE_FLOW,
+	get title() {
+		return translate( 'VideoPress TV' );
+	},
+	useSteps() {
+		return [
+			{ slug: 'processing', component: ProcessingStep },
+			{ slug: 'redirect', component: VideoPressTvRedirectStep },
+		];
+	},
+
+	useStepNavigation( _currentStep, navigate ) {
+		if ( document.body ) {
+			// Make sure we only target videopress tv stepper for body css
+			if ( ! document.body.classList.contains( 'is-videopress-tv-purchase-stepper' ) ) {
+				document.body.classList.add( 'is-videopress-tv-purchase-stepper' );
+			}
+		}
+
+		const name = this.name;
+		const locale = useLocale();
+		const { setPendingAction, setProgress, setSelectedSite } = useDispatch( ONBOARD_STORE );
+		const { setIntentOnSite } = useDispatch( SITE_STORE );
+		const { supportedPlans } = useSupportedPlans( locale, 'MONTHLY' );
+		const _siteSlug = useSiteSlug();
+		const userIsLoggedIn = useSelect(
+			( select ) => ( select( USER_STORE ) as UserSelect ).isCurrentUserLoggedIn(),
+			[]
+		);
+		const userData = useSelect(
+			( select ) => ( select( USER_STORE ) as UserSelect ).getCurrentUser(),
+			[]
+		);
+		const siteData =
+			useSelect(
+				( select ) => ( select( SITE_STORE ) as SiteSelect ).getSite( _siteSlug || '' ),
+				[]
+			) || {};
+		const getPlanProduct = useSelect(
+			( select ) => ( select( PLANS_STORE ) as PlansSelect ).getPlanProduct,
+			[]
+		);
+
+		const [ isVideoPressTvPurchasePending, setIsVideoPressTvPurchasePending ] = useState( false );
+
+		const addVideoPressPendingAction = () => {
+			// if the supported plans haven't been received yet, wait for next rerender to try again.
+			if ( 0 === supportedPlans.length ) {
+				return;
+			}
+
+			// If the site is already on a plan, we don't need to add a pending action. Maybe redirect to site?
+			const siteFreePlan = siteData?.plan?.is_free;
+			if ( ! siteFreePlan ) {
```

Comment:
```suggestion
			if ( false === siteFreePlan ) {
```

Reply:
If siteData or plan is null, then it should probably be considered site with no plan ?

Reply:
Thanks, will add separately!